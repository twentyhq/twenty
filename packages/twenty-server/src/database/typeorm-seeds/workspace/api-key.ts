import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';

const tableName = 'apiKey';

const API_KEY_ID = '20202020-f401-4d8a-a731-64d007c27bad';

export const seedApiKey = async (
  entityManager: WorkspaceEntityManager,
  schemaName: string,
) => {
  await entityManager
    .createQueryBuilder(undefined, undefined, undefined, {
      shouldBypassPermissionChecks: true,
    })
    .insert()
    .into(`${schemaName}.${tableName}`, ['id', 'name', 'expiresAt'])
    .orIgnore()
    .values([
      {
        id: API_KEY_ID,
        name: 'My api key',
        expiresAt: new Date(
          new Date().getTime() + 1000 * 60 * 60 * 24 * 365 * 100, // In 100 years
        ),
      },
    ])
    .execute();
};
