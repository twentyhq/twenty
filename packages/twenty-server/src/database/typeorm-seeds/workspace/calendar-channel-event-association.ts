import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';

const tableName = 'calendarChannelEventAssociation';

export const seedCalendarChannelEventAssociations = async (
  entityManager: WorkspaceEntityManager,
  schemaName: string,
) => {
  await entityManager
    .createQueryBuilder(undefined, undefined, undefined, {
      shouldBypassPermissionChecks: true,
    })
    .insert()
    .into(`${schemaName}.${tableName}`, [
      'id',
      'calendarChannelId',
      'calendarEventId',
      'eventExternalId',
      'recurringEventExternalId',
    ])
    .orIgnore()
    .values([
      {
        id: 'e1ab9e1b-df6e-438e-a788-11c96dcecdd3',
        calendarChannelId: '59efdefe-a40f-4faf-bb9f-c6f9945b8203',
        calendarEventId: '86083141-1c0e-494c-a1b6-85b1c6fefaa5',
        eventExternalId: 'exampleExternalId',
        recurringEventExternalId: 'exampleRecurringExternalId',
      },
    ])
    .execute();
};
