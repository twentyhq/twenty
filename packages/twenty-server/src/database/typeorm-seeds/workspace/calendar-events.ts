import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';

const tableName = 'calendarEvent';

export const DEV_SEED_CALENDAR_EVENT_IDS = {
  CALENDAR_EVENT_1: '86083141-1c0e-494c-a1b6-85b1c6fefaa5',
};

export const seedCalendarEvents = async (
  entityManager: WorkspaceEntityManager,
  schemaName: string,
) => {
  await entityManager
    .createQueryBuilder(undefined, undefined, undefined, {
      shouldBypassPermissionChecks: true,
    })
    .insert()
    .into(`${schemaName}.${tableName}`, [
      'id',
      'title',
      'isCanceled',
      'isFullDay',
      'startsAt',
      'endsAt',
      'externalCreatedAt',
      'externalUpdatedAt',
      'description',
      'location',
      'iCalUID',
      'conferenceSolution',
      'conferenceLinkPrimaryLinkLabel',
      'conferenceLinkPrimaryLinkUrl',
    ])
    .orIgnore()
    .values([
      {
        id: DEV_SEED_CALENDAR_EVENT_IDS.CALENDAR_EVENT_1,
        title: 'Meeting with Christoph',
        isCanceled: false,
        isFullDay: false,
        startsAt: new Date(new Date().setHours(10, 0)).toISOString(),
        endsAt: new Date(new Date().setHours(11, 0)).toISOString(),
        externalCreatedAt: new Date().toISOString(),
        externalUpdatedAt: new Date().toISOString(),
        description: 'Discuss project progress',
        location: 'Seattle',
        iCalUID: 'event1@calendar.com',
        conferenceSolution: 'Zoom',
        conferenceLinkPrimaryLinkLabel: 'https://zoom.us/j/1234567890',
        conferenceLinkPrimaryLinkUrl: 'https://zoom.us/j/1234567890',
      },
    ])
    .execute();
};
