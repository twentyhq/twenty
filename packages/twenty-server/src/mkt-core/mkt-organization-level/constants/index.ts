export * from './permission-templates.constants';
export * from './rbac-definitions.constants';
export * from './validation.constants';

export enum OrganizationHierarchyLevel {
  MIN = 1,
  MAX = 8,
}
