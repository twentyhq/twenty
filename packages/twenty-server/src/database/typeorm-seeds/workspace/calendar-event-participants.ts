import { DEV_SEED_PERSON_IDS } from 'src/database/typeorm-seeds/workspace/seedPeople';
import { DEV_SEED_WORKSPACE_MEMBER_IDS } from 'src/database/typeorm-seeds/workspace/workspace-members';
import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';
import { CalendarEventParticipantResponseStatus } from 'src/modules/calendar/common/standard-objects/calendar-event-participant.workspace-entity';

const tableName = 'calendarEventParticipant';

export const seedCalendarEventParticipants = async (
  entityManager: WorkspaceEntityManager,
  schemaName: string,
) => {
  await entityManager
    .createQueryBuilder(undefined, undefined, undefined, {
      shouldBypassPermissionChecks: true,
    })
    .insert()
    .into(`${schemaName}.${tableName}`, [
      'id',
      'calendarEventId',
      'handle',
      'displayName',
      'isOrganizer',
      'responseStatus',
      'personId',
      'workspaceMemberId',
    ])
    .orIgnore()
    .values([
      {
        id: 'da8f47c3-8055-49ad-b7e4-9c9d5bbc1ecc',
        calendarEventId: '86083141-1c0e-494c-a1b6-85b1c6fefaa5',
        handle: 'christoph.calisto@linkedin.com',
        displayName: 'Christoph Calisto',
        isOrganizer: true,
        responseStatus: CalendarEventParticipantResponseStatus.ACCEPTED,
        personId: DEV_SEED_PERSON_IDS.CHRISTOPH,
        workspaceMemberId: null,
      },
      {
        id: 'e1ab9e1b-df6e-438e-a788-11c96dcecdd3',
        calendarEventId: '86083141-1c0e-494c-a1b6-85b1c6fefaa5',
        handle: 'tim@apple.com',
        displayName: 'Tim Apple',
        isOrganizer: false,
        responseStatus: CalendarEventParticipantResponseStatus.ACCEPTED,
        personId: null,
        workspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
      },
    ])
    .execute();
};
