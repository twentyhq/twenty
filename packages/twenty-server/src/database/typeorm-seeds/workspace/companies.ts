import { DEV_SEED_WORKSPACE_MEMBER_IDS } from 'src/database/typeorm-seeds/workspace/workspace-members';
import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';

const tableName = 'company';

export const DEV_SEED_COMPANY_IDS = {
  LINKEDIN: '20202020-3ec3-4fe3-8997-b76aa0bfa408',
  FACEBOOK: '20202020-5d81-46d6-bf83-f7fd33ea6102',
  QONTO: '20202020-0713-40a5-8216-82802401d33e',
  MICROSOFT: '20202020-ed89-413a-b31a-962986e67bb4',
  AIRBNB: '20202020-171e-4bcc-9cf7-43448d6fb278',
  GOOGLE: '20202020-c21e-4ec2-873b-de4264d89025',
  NETFLIX: '20202020-707e-44dc-a1d2-30030bf1a944',
  LIBEO: '20202020-3f74-492d-a101-2a70f50a1645',
  CLAAP: '20202020-cfbf-4156-a790-e39854dcd4eb',
  HASURA: '20202020-f86b-419f-b794-02319abe8637',
  WEWORK: '20202020-5518-4553-9433-42d8eb82834b',
  SAMSUNG: '20202020-f79e-40dd-bd06-c36e6abb4678',
  ALGOLIA: '20202020-1455-4c57-afaf-dd5dc086361d',
};

export const seedCompanies = async (
  entityManager: WorkspaceEntityManager,
  schemaName: string,
) => {
  await entityManager
    .createQueryBuilder(undefined, undefined, undefined, {
      shouldBypassPermissionChecks: true,
    })
    .insert()
    .into(`${schemaName}.${tableName}`, [
      'id',
      'name',
      'domainNamePrimaryLinkUrl',
      'addressAddressStreet1',
      'addressAddressStreet2',
      'addressAddressCity',
      'addressAddressState',
      'addressAddressPostcode',
      'addressAddressCountry',
      'position',
      'tagline',
      'workPolicy',
      'createdBySource',
      'createdByWorkspaceMemberId',
      'createdByName',
    ])
    .orIgnore()
    .values([
      {
        id: DEV_SEED_COMPANY_IDS.LINKEDIN,
        name: 'Linkedin',
        domainNamePrimaryLinkUrl: 'https://linkedin.com',
        addressAddressStreet1: 'Eutaw Street',
        addressAddressStreet2: null,
        addressAddressCity: 'Dublin',
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: 'Ireland',
        position: 1,
        tagline: "Linkedin's tagline",
        workPolicy: ['ON_SITE'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.FACEBOOK,
        name: 'Facebook',
        domainNamePrimaryLinkUrl: 'https://facebook.com',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 2,
        tagline: "Facebook's tagline",
        workPolicy: ['REMOTE_WORK'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.QONTO,
        name: 'Qonto',
        domainNamePrimaryLinkUrl: 'https://qonto.com',
        addressAddressStreet1: '18 rue de navarrin',
        addressAddressStreet2: null,
        addressAddressCity: 'Paris',
        addressAddressState: null,
        addressAddressPostcode: '75009',
        addressAddressCountry: 'France',
        position: 3,
        tagline: "Qonto's tagline",
        workPolicy: ['ON_SITE'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.MICROSOFT,
        name: 'Microsoft',
        domainNamePrimaryLinkUrl: 'https://microsoft.com',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 4,
        tagline: "Microsoft's tagline",
        workPolicy: ['ON_SITE', 'HYBRID'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.AIRBNB,
        name: 'Airbnb',
        domainNamePrimaryLinkUrl: 'https://airbnb.com',
        addressAddressStreet1: '888 Brannan St',
        addressAddressStreet2: null,
        addressAddressCity: 'San Francisco',
        addressAddressState: 'CA',
        addressAddressPostcode: '94103',
        addressAddressCountry: 'United States',
        position: 5,
        tagline: "Airbnb's tagline",
        workPolicy: ['ON_SITE'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.GOOGLE,
        name: 'Google',
        domainNamePrimaryLinkUrl: 'https://google.com',
        addressAddressStreet1: '760 Market St',
        addressAddressStreet2: 'Floor 10',
        addressAddressCity: 'San Francisco',
        addressAddressState: null,
        addressAddressPostcode: '94102',
        addressAddressCountry: 'United States',
        position: 6,
        tagline: "Google's tagline",
        workPolicy: ['HYBRID', 'REMOTE_WORK'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.NETFLIX,
        name: 'Netflix',
        domainNamePrimaryLinkUrl: 'https://netflix.com',
        addressAddressStreet1: '2300 Harrison St',
        addressAddressStreet2: null,
        addressAddressCity: 'San Francisco',
        addressAddressState: 'CA',
        addressAddressPostcode: '94110',
        addressAddressCountry: 'United States',
        position: 7,
        tagline: "Netflix's tagline",
        workPolicy: ['REMOTE_WORK'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.LIBEO,
        name: 'Libeo',
        domainNamePrimaryLinkUrl: 'https://libeo.io',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 8,
        tagline: "Libeo's tagline",
        workPolicy: ['HYBRID'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.CLAAP,
        name: 'Claap',
        domainNamePrimaryLinkUrl: 'https://claap.io',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 9,
        tagline: "Claap's tagline",
        workPolicy: ['ON_SITE', 'REMOTE_WORK'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.HASURA,
        name: 'Hasura',
        domainNamePrimaryLinkUrl: 'https://hasura.io',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 10,
        tagline: "Hasura's tagline",
        workPolicy: ['HYBRID'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.WEWORK,
        name: 'Wework',
        domainNamePrimaryLinkUrl: 'https://wework.com',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 11,
        tagline: "Wework's tagline",
        workPolicy: ['ON_SITE'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.SAMSUNG,
        name: 'Samsung',
        domainNamePrimaryLinkUrl: 'https://samsung.com',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 12,
        tagline: "Samsung's tagline",
        workPolicy: ['HYBRID'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
      {
        id: DEV_SEED_COMPANY_IDS.ALGOLIA,
        name: 'Algolia',
        domainNamePrimaryLinkUrl: 'https://algolia.com',
        addressAddressStreet1: null,
        addressAddressStreet2: null,
        addressAddressCity: null,
        addressAddressState: null,
        addressAddressPostcode: null,
        addressAddressCountry: null,
        position: 13,
        tagline: "Algolia's tagline",
        workPolicy: ['ON_SITE'],
        createdBySource: 'MANUAL',
        createdByWorkspaceMemberId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        createdByName: 'Tim Apple',
      },
    ])
    .execute();
};
