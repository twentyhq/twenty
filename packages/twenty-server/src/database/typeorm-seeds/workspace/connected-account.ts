import { DEV_SEED_WORKSPACE_MEMBER_IDS } from 'src/database/typeorm-seeds/workspace/workspace-members';
import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';

const tableName = 'connectedAccount';

export const DEV_SEED_CONNECTED_ACCOUNT_IDS = {
  TIM: '20202020-9ac0-4390-9a1a-ab4d2c4e1bb7',
  JONY: '20202020-0cc8-4d60-a3a4-803245698908',
  PHIL: '20202020-cafc-4323-908d-e5b42ad69fdf',
};

export const seedConnectedAccount = async (
  entityManager: WorkspaceEntityManager,
  schemaName: string,
) => {
  await entityManager
    .createQueryBuilder(undefined, undefined, undefined, {
      shouldBypassPermissionChecks: true,
    })
    .insert()
    .into(`${schemaName}.${tableName}`, [
      'id',
      'createdAt',
      'updatedAt',
      'deletedAt',
      'lastSyncHistoryId',
      'accountOwnerId',
      'refreshToken',
      'accessToken',
      'provider',
      'handle',
    ])
    .orIgnore()
    .values([
      {
        id: DEV_SEED_CONNECTED_ACCOUNT_IDS.TIM,
        createdAt: new Date(),
        updatedAt: new Date(),
        deletedAt: null,
        lastSyncHistoryId: 'exampleLastSyncHistory',
        accountOwnerId: DEV_SEED_WORKSPACE_MEMBER_IDS.TIM,
        refreshToken: 'exampleRefreshToken',
        accessToken: 'exampleAccessToken',
        provider: 'google',
        handle: 'tim@apple.dev',
      },
      {
        id: DEV_SEED_CONNECTED_ACCOUNT_IDS.JONY,
        createdAt: new Date(),
        updatedAt: new Date(),
        deletedAt: null,
        lastSyncHistoryId: 'exampleLastSyncHistory',
        accountOwnerId: DEV_SEED_WORKSPACE_MEMBER_IDS.JONY,
        refreshToken: 'exampleRefreshToken',
        accessToken: 'exampleAccessToken',
        provider: 'google',
        handle: 'jony.ive@apple.dev',
      },
      {
        id: DEV_SEED_CONNECTED_ACCOUNT_IDS.PHIL,
        createdAt: new Date(),
        updatedAt: new Date(),
        deletedAt: null,
        lastSyncHistoryId: 'exampleLastSyncHistory',
        accountOwnerId: DEV_SEED_WORKSPACE_MEMBER_IDS.PHIL,
        refreshToken: 'exampleRefreshToken',
        accessToken: 'exampleAccessToken',
        provider: 'google',
        handle: 'phil.schiler@apple.dev',
      },
    ])
    .execute();
};
