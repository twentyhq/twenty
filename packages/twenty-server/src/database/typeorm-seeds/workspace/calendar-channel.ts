import { DEV_SEED_CONNECTED_ACCOUNT_IDS } from 'src/database/typeorm-seeds/workspace/connected-account';
import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';
import { CalendarChannelVisibility } from 'src/modules/calendar/common/standard-objects/calendar-channel.workspace-entity';

const tableName = 'calendarChannel';

export const seedCalendarChannels = async (
  entityManager: WorkspaceEntityManager,
  schemaName: string,
) => {
  await entityManager
    .createQueryBuilder(undefined, undefined, undefined, {
      shouldBypassPermissionChecks: true,
    })
    .insert()
    .into(`${schemaName}.${tableName}`, [
      'id',
      'connectedAccountId',
      'handle',
      'visibility',
      'isContactAutoCreationEnabled',
      'isSyncEnabled',
    ])
    .orIgnore()
    .values([
      {
        id: '59efdefe-a40f-4faf-bb9f-c6f9945b8203',
        connectedAccountId: DEV_SEED_CONNECTED_ACCOUNT_IDS.TIM,
        handle: 'tim@apple.com',
        visibility: CalendarChannelVisibility.SHARE_EVERYTHING,
        isContactAutoCreationEnabled: true,
        isSyncEnabled: true,
      },
    ])
    .execute();
};
