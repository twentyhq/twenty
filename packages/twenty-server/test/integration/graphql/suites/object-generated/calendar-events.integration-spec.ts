import { CALENDAR_EVENT_GQL_FIELDS } from 'test/integration/constants/calendar-event-gql-fields.constants';
import { findManyOperationFactory } from 'test/integration/graphql/utils/find-many-operation-factory.util';
import { findOneOperationFactory } from 'test/integration/graphql/utils/find-one-operation-factory.util';
import { makeGraphqlAPIRequest } from 'test/integration/graphql/utils/make-graphql-api-request.util';

import { DEV_SEED_CALENDAR_EVENT_IDS } from 'src/database/typeorm-seeds/workspace/calendar-events';

describe('calendarEventsResolver (e2e)', () => {
  it('should find many calendarEvents', async () => {
    const graphqlOperation = findManyOperationFactory({
      objectMetadataSingularName: 'calendarEvent',
      objectMetadataPluralName: 'calendarEvents',
      gqlFields: CALENDAR_EVENT_GQL_FIELDS,
    });

    const response = await makeGraphqlAPIRequest(graphqlOperation);

    const data = response.body.data.calendarEvents;

    expect(data).toBeDefined();
    expect(Array.isArray(data.edges)).toBe(true);

    const edges = data.edges;

    expect(edges.length).toEqual(1);

    const calendarEvent = edges[0].node;

    expect(calendarEvent).toMatchSnapshot({
      createdAt: expect.any(String),
      endsAt: expect.any(String),
      startsAt: expect.any(String),
      updatedAt: expect.any(String),
    });
  });

  it('should find one calendarEvent', async () => {
    const graphqlOperation = findOneOperationFactory({
      objectMetadataSingularName: 'calendarEvent',
      filter: { id: { eq: DEV_SEED_CALENDAR_EVENT_IDS.CALENDAR_EVENT_1 } },
      gqlFields: CALENDAR_EVENT_GQL_FIELDS,
    });

    const response = await makeGraphqlAPIRequest(graphqlOperation);

    const data = response.body.data.calendarEvent;

    expect(data).toBeDefined();
    expect(data).toMatchSnapshot({
      createdAt: expect.any(String),
      endsAt: expect.any(String),
      startsAt: expect.any(String),
      updatedAt: expect.any(String),
    });
  });
});
