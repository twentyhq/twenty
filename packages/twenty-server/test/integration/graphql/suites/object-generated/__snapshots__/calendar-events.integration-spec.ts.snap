// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendarEventsResolver (e2e) should find many calendarEvents 1`] = `
{
  "createdAt": Any<String>,
  "deletedAt": null,
  "description": "Discuss project progress",
  "endsAt": Any<String>,
  "id": "86083141-1c0e-494c-a1b6-85b1c6fefaa5",
  "startsAt": Any<String>,
  "title": "Meeting with Christoph",
  "updatedAt": Any<String>,
}
`;

exports[`calendarEventsResolver (e2e) should find one calendarEvent 1`] = `
{
  "createdAt": Any<String>,
  "deletedAt": null,
  "description": "Discuss project progress",
  "endsAt": Any<String>,
  "id": "86083141-1c0e-494c-a1b6-85b1c6fefaa5",
  "startsAt": Any<String>,
  "title": "Meeting with Christoph",
  "updatedAt": Any<String>,
}
`;
